app/javascript/utils/utils.js

```diff
@@ -21,6 +21,28 @@ export const formatPercent = (value) => {
   return ((value * 100).toFixed(0) + '%')
 }
 
+export const filterChanged = (prev, current) => {
+  if (!prev) {
+    return current
+  }
+  const changedFields = {}
+  forEach(current, (value, key) => {
+    if (!isEqual(prev[key], value)) {
+      if (isPlainObject(value) && prev[key] !== null) {
+        const obj = filterChanged(prev[key], value)
+        if (!isEmpty(obj)) {
+          obj['id'] = value.id
```

Comment:
I don't think this case can happen in current scenario as this is used only for editing defined object. Or more precisely for objects related to `application`, like `primary_applicant`. If we were to add element to an array it would be handled by line 38. But if you like I can future proof it. But should id be set to `null` in that case or not added at all?

Reply:
I'd say it should not be added at all, but you're right that practically this is unlikely

Reply:
what should I do? Leave it as is, future proof or sth else?

Reply:
I think we should future-proof it. I can see a scenario where a nested object doesn't have an id in the future